purenix/src/Lib.hs

```diff
@@ -2,58 +2,27 @@
 
 module Lib where
 
-import Nix.Prelude
-
-import Data.Aeson (decode)
+import qualified Data.Aeson as Aeson
 import Data.Aeson.Types (parseEither)
 import qualified Data.Text as T
-import Data.Text.Lazy (pack)
-import Data.Text.Lazy.Encoding (encodeUtf8)
 import qualified Data.Text.Lazy.IO as TL
-import Language.PureScript.CoreFn (modulePath)
 import Language.PureScript.CoreFn.FromJSON (moduleFromJSON)
 import Nix.Convert (convert)
+import Nix.Prelude
 import Nix.Print (renderExpr)
-import Nix.Util (stripAnnMod)
-import System.Directory (createDirectoryIfMissing)
-import Text.Pretty.Simple (pPrint)
-import System.Directory (copyFile, doesFileExist)
-import qualified System.Environment as Env
+import qualified System.Directory as Dir
 import qualified System.Exit as Sys
-import System.FilePath (replaceExtension)
+import System.FilePath ((</>))
 
 defaultMain :: IO ()
 defaultMain = do
-  Env.getArgs >>= print
-  let modName = "Main"
-      moduleDir = "../purescript-cabal-parser/output/" <> modName <> "/"
-      corefnFilePath = moduleDir <> "corefn.json"
-  corefn <- readFile corefnFilePath
-  -- TODO why is this lazy?
-  let maybeValue = decode $ encodeUtf8 $ pack corefn
-  case maybeValue of
-    Nothing -> error "wasn't able to decode corefn as json"
-    Just value -> do
-      let eitherModule = parseEither moduleFromJSON value
-      case eitherModule of
-        Left err -> error err
-        Right (_version, mdl) -> do
-          putStrLn "successfully decoded purescript module:"
-          pPrint (stripAnnMod mdl)
-          let generateDir = "../generated/"
-          createDirectoryIfMissing True generateDir
-
-          -- Create the output __ffi.nix file for this module if it has foreign imports.
-          let ffiFilePath = "../purescript-cabal-parser/" <> replaceExtension (modulePath mdl) "nix"
-          let ffiOutputFilePath = generateDir <> modName <> "__ffi.nix"
-          doesFfiFileExist <- doesFileExist ffiFilePath
-          when doesFfiFileExist $ copyFile ffiFilePath ffiOutputFilePath
```

Comment:
> Main.nix doesn't actually get copied over into the output directory. That seems like an issue to me, I thought it was spago's responsibility to copy that? I feel like you explained this but then I forgot.

Ah yeah, I was just commenting about this.

I think it is the responsibility of `purenix` to copy over the ffi file.  `spago` doesn't help us here.

This block of code is what was responsible for copying over the FFI file.

You can find the path of the input file using [`modulePath`](https://hackage.haskell.org/package/purescript-0.13.8/docs/Language-PureScript-CoreFn-Module.html), and then just replace the extension.  If that file exists, it needs to be copied over and renamed.